### DISCUSSION TITLE
Support for plain PKCE challenges

### DISCUSSION THREAD

**Discussion by Pelps12** [panva/oauth4webapi#43] [Q&A] [answered]
Would it be possible to add plain PKCE Code challenge generation? Some "legacy" APIs may not support it. I could create a pull request for it. (Would be my first PR though).

**Comment by panva** [comment, marked as answer]
But, ... it's plain, there's already a method for generating a verifier. 

What else do you imagine needs to be done?

**Comment by Pelps12** [comment]
I'm sorry if I'm getting something wrong. The PKCEcodeChallenge  function only creates a challenge with SHA-256 but I've seen some authorization server that still use the plain version

**Reply by panva** [reply]
Ok. But then there's nothing to be done. plain just means the challenge is the same as the verifier.

**Comment by Pelps12** [comment]
5 hours of my life wasted 😭. Sorry for wasting your time

**Reply by panva** [reply]
Sorry about that ;) please keep in mind plain [PKCE](https://www.rfc-editor.org/rfc/rfc7636.html) doesn't have the same protections. It's truly legacy and not considered when it comes to the profiles this library declares support for.

**Reply by Pelps12** [reply]
I understand the risks associated (kinda). The services I need to use don't have the best OAuth implementations (cough cough MyAnimeList). SHA-256 doesn't even seem that hard to implement (may be wrong).
